src/main/java/it/pagopa/interop/probing/eservice/operations/repository/EserviceRepository.java

```diff
@@ -1,16 +1,22 @@
 package it.pagopa.interop.probing.eservice.operations.repository;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
-
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
-
+import it.pagopa.interop.probing.eservice.operations.dtos.SearchProducerNameResponse;
 import it.pagopa.interop.probing.eservice.operations.model.Eservice;
 
 @Repository
 public interface EserviceRepository extends JpaRepository<Eservice, Long> {
-	Optional<Eservice> findByEserviceIdAndVersionId(@Param("eserviceId") UUID eserviceId,
-			@Param("versionId") UUID versionId);
+  Optional<Eservice> findByEserviceIdAndVersionId(@Param("eserviceId") UUID eserviceId,
+      @Param("versionId") UUID versionId);
+
+  @Query("SELECT DISTINCT new it.pagopa.interop.probing.eservice.operations.dtos.SearchProducerNameResponse(e.producerName, e.producerName) FROM Eservice e WHERE UPPER(e.producerName) LIKE %:producerName%")
```

Comment:
You can use CriteriaBuilder to achieve that, so you can avoid to write SQL as String

Reply:
Done